src/makeCalculator.js

```diff
@@ -4,7 +4,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    add(number) {
+      this.result += number;
+    },
+
+    subtract(number) {
+      this.result -= number;
+    },
+
+    multiply(number) {
+      this.result *= number;
+    },
+
+    divide(number) {
+      if (number === 0) {
+        throw new Error('Cannot divide by zero');
+      }
+
+      this.result /= number;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(callback, number) {
+      if (typeof callback === 'function') {
+        callback.call(this, number);
+      }
+
+      return this;
+    },
+  };
+
+  return calculator;
 }
```

Comment:
The function `makeCalculator` is correctly named according to the task description. However, the methods within the `calculator` object should use the method shorthand syntax instead of the `function` keyword.